### ISSUE TITLE
o-header: add missing focus state to the international edition link - ID: DAC_Focus_Visible_02

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#548]
# Issue
This is from page 166 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-header@9.0.1&demo=header&system_code=origami-repo-data&brand=master>

When the side navigation menu is activated, the ‘Switch to International Edition’ link has no visible focus indicator. This issue may affect low-vision and keyboard alone users who may find difficulty in distinguishing when focus has been placed on the link

# Proposed Solution:
Ensure all focusable elements have a visible focus highlighting mechanism
